### PR TITLE
feat: pull add on images through replicated proxy

### PR DESCRIPTION
**Pull request by ricardomaraschini**
replace add on image references in order to pull images using replicated proxy for oci artifacts.

### CODE REVIEW

File: Makefile
Comment (ricardomaraschini) [inline Makefile:49]: small indentation problem here.

File: pkg/addons/velero/velero.go
Comment (ricardomaraschini) [inline pkg/addons/velero/velero.go:61]: this is on me, i don't think we need the `docker.io` prefix here.

File: cmd/buildtools/openebs.go
Comment (banjoh) [inline cmd/buildtools/openebs.go:95]: I'd be confused by this print statement where a number just gets printed

File: cmd/buildtools/openebs.go
Comment (banjoh) [inline cmd/buildtools/openebs.go:121]: ...and this one

File: cmd/buildtools/seaweedfs.go
Comment (banjoh) [inline cmd/buildtools/seaweedfs.go:53]: and this one

File: Makefile
Comment (ricardomaraschini) [inline Makefile:15]: This caught my eye, I looked into the helm chart and it seems to me that this should be `4.1.0` instead of `4.0.0`.

File: Makefile
Comment (ricardomaraschini) [inline Makefile:14]: Somehow this seems to be downgrading OpenEBS. This does not feel right, maybe this is from before `4.1.0` became available.

File: cmd/buildtools/helm.go
Comment (ricardomaraschini) [inline cmd/buildtools/helm.go:225]: I think we are using `fmt.Errorf` everywhere so maybe we should use it on this file as well.

File: cmd/buildtools/helm.go
Comment (ricardomaraschini) [inline cmd/buildtools/helm.go:201]: Maybe we should use 1.28 as this is the oldest ec version we have. I recognise that I don't fully understand the impact of this, I assume it will render only types that are known to 1.28 (maybe do some conversion?). Is that it ?

File: cmd/buildtools/helm.go
Comment (ricardomaraschini) [inline cmd/buildtools/helm.go:243]: IIUC we are sorta removing all spaces from `rel.Manifest` but we don't do the same with `m.Manifest`. Maybe the why will become clear as I keep reading the code.

File: cmd/buildtools/utils.go
Comment (ricardomaraschini) [inline cmd/buildtools/utils.go:123]: Can we return an error here ?

File: cmd/buildtools/utils.go
Comment (ricardomaraschini) [inline cmd/buildtools/utils.go:125]: Can we receive this context as a function argument instead ?

File: pkg/addons/seaweedfs/seaweedfs.go
Comment (ricardomaraschini) [inline pkg/addons/seaweedfs/seaweedfs.go:74]: Doesn't this always evaluate to true ?

File: Makefile
Comment (ricardomaraschini) [inline Makefile:27]: It seems the automation erroneously bumped this in the past. The latest registry chart (`v2.2.3`) ships with registry `2.8.1`.  I don't expect to see any problems once this is in I need to test an upgrade just to be sure.

File: Makefile
Comment (ricardomaraschini) [inline Makefile:78]: I am inclined to make the automation to write this down to a `json` file in each add-on directory that then can be embedded. What do you think ? We would not need to have any of these tags.

File: Makefile
Comment (laverya) [reply Makefile:27]: I would strongly prefer to keep using the 2.8.3 image version

File: Makefile
Comment (emosbaugh) [reply Makefile:78]: along the same lines it would be pretty cool since buildtools discovers images in charts if it were to generate the values yaml file in each add-on directory to override images

**Comment by emosbaugh** [comment]
lgtm pending andrew's comment and merge of the other pr

File: pkg/addons/adminconsole/static/values.tpl.yaml
Comment (emosbaugh) [inline pkg/addons/adminconsole/static/values.tpl.yaml:2]: I think you missed my point here but I realize what I said was very unclear. Right now there is a hardcoded list of images in this values file. But your new code discovers images in helm charts. What if a new image gets added? I thought it may be cool to dynamically generate the helm values.yaml overriding images so we would automatically proxy new images. That way we would never make a release with an image we didn't know about and therefore break our rules around whitelisted domains. It was just a thought and I'm happy to approve this pull request as is.

File: pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go
Comment (emosbaugh) [inline pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go:146]: Do you really need a loop here since this is a map?

File: pkg/addons/adminconsole/static/values.tpl.yaml
Comment (emosbaugh) [reply pkg/addons/adminconsole/static/values.tpl.yaml:2]: Oh I see why this does not make sense now. Since you are parsing the rendered yaml and not the values files. But I do wonder if we could compare the list of discovered images with the list of expect images and somehow warn.

Comment (emosbaugh) [reply pkg/addons/adminconsole/static/values.tpl.yaml:2]: oh i see this is not possible since you are extracting images from the rendered yaml but not the values files. as a follow up it may be nice to provide a sanity check, since we have a hardcoded list of images we are expecting, and if we find are any images we do not know about in the rendered template produce an error.

**Review by emosbaugh — APPROVED** [review]

File: Makefile
Comment (ricardomaraschini) [reply Makefile]: done.

File: pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go
Comment (ricardomaraschini) [reply pkg/addons/embeddedclusteroperator/embeddedclusteroperator.go]: No, that is not necessary. Solved.

File: cmd/buildtools/utils.go
Comment (sgalsaleh) [inline cmd/buildtools/utils.go:315]: typo
```suggestion
type ReducedNestedSpec struct {
```

File: cmd/buildtools/velero.go
Comment (sgalsaleh) [inline cmd/buildtools/velero.go:20]: is this something that should be done now or later at some point?

File: pkg/addons/openebs/static/metadata.yaml
Comment (sgalsaleh) [inline pkg/addons/openebs/static/metadata.yaml:9]: this image location uses ttl.sh, this doesn't seem expected

File: pkg/addons/openebs/static/metadata.yaml
Comment (sgalsaleh) [inline pkg/addons/openebs/static/metadata.yaml:10]: i assume these are all the images found in the chart, but not necessarily the ones we actually end up using?

File: pkg/addons/openebs/static/metadata.yaml
Comment (ricardomaraschini) [reply pkg/addons/openebs/static/metadata.yaml:9]: These references to ttl are there for testing purposes. :-) Let me clean this up.

File: cmd/buildtools/velero.go
Comment (ricardomaraschini) [reply cmd/buildtools/velero.go:20]: Yeah, this should be removed before merging.

File: pkg/addons/openebs/static/metadata.yaml
Comment (sgalsaleh) [reply pkg/addons/openebs/static/metadata.yaml:10]: no problem. why are there so many images? the airgap builder doesn't include all of these images in the airgap bundle.

Comment (ricardomaraschini) [reply pkg/addons/openebs/static/metadata.yaml:10]: solved as discussed.

File: cmd/buildtools/velero.go
Comment (ricardomaraschini) [reply cmd/buildtools/velero.go]: removed.

File: cmd/buildtools/utils.go
Comment (ricardomaraschini) [reply cmd/buildtools/utils.go]: done.

**Review by emosbaugh — APPROVED** [review]

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**
